src/common/dao/immutable.go

```diff
@@ -0,0 +1,84 @@
+package dao
+
+import (
+	"fmt"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/common/models"
+)
+
+// CreateImmutableRule creates the Immutable Rule
+func CreateImmutableRule(ir *models.ImmutableRule) (int64, error) {
+	ir.Enabled = true
+	if exceedMaxImmutableRuleCount(ir.ProjectID) {
+		return 0, fmt.Errorf("immutable tag rule count exceed %v", common.MaxRuleCount)
+	}
```

Comment:
I think the checking rule count should be handled in the upper level. The dao should only handle the data access related operations

Reply:
Removed